api/controllers/bookmark.js

```diff
@@ -29,33 +31,42 @@ let log = require("winston");
  * @apiError NotAuthorized The user doesn't have permission to perform this operation.
  *
  */
-router.get("/list/:userId", function(req, res, next) {
-  // Find out if userID exists in user table
-  try {
-    let userId = req.params.userId;
-    db
-      .any("SELECT * FROM bookmarks WHERE userid=$1", [userId])
-      .then(data => {
-        res.json({
-          success: true,
-          status: "success",
-          data: data,
-          message: "Retrieved ALL users"
-        });
-      })
-      .catch(function(err) {
-        res.json({
-          success: false,
-          error: error.message || error
-        });
-        log.error(err);
+
+function lookupBookmarksById(req, res, userId, next) {
+  db
+    .any("SELECT * FROM bookmarks WHERE userid=$1", [userId])
+    .then(data => {
+      res.json({
+        success: true,
+        status: "success",
+        data: data,
+        message: "Retrieved ALL bookmarks for specified user"
       });
-  } catch (err) {
-    console.error(err);
-    log.error(err);
-    return next(err);
+    })
+    .catch(function(error) {
+      res.json({
+        success: false,
+        error: error.message || error
+      });
+      log.error(error);
+    });
+}
+
+function queryBookmarks(req, res, next) {
+  let userid = req.params.userid;
+
+  if (!userid) {
+    ensureUser(req, res, function(req, res, next) {
```

Comment:
Nevermind, I guess the idea here is that you can look at the bookmarks of any user, not just your own?